aridanalysis/aridanalysis.py

```diff
@@ -282,40 +287,116 @@ def arid_logreg(data_frame, response, features=[], type="binomial", model="addit
+    assert isinstance(cat_features, list), "ERROR: INVALID LIST INTPUT PASSED"
+    
+    
+    #Deal with the features column
+    if len(con_features) == 0:
+        con_features = data_frame.drop(columns=[response]).select_dtypes('number').columns.tolist()
+    if len(cat_features) == 0:
+        cat_features = data_frame.drop(columns=[response]).select_dtypes(['category', 'object']).columns.tolist()
+    
+    
+    assert isinstance(data_frame, pd.DataFrame), errors.INVALID_DATAFRAME
+    assert not data_frame.empty , errors.EMPTY_DATAFRAME
+    assert response in data_frame.columns.tolist(), errors.RESPONSE_NOT_FOUND
+    assert all(item in data_frame.columns.tolist() for item in con_features), "ERROR: CONTINUOUS VARIABLE(S) NOT IN DATAFRAME"
+    assert all(item in data_frame.columns.tolist() for item in cat_features), "ERROR: CATEGORICAL VARIABLE(S) NOT IN DATAFRAME"
+    assert ptypes.is_integer_dtype(data_frame[response].dtype), "ERROR: INVALID RESPONSE DATATYPE FOR COUNT REGRESSION: MUST BE TYPE INT"
+    assert model in ["additive", "interactive"], "ERROR: INVALID MODEL PASSED"
+    assert ptypes.is_numeric_dtype(type(alpha)), errors.INVALID_ALPHA_INPUT
+    
+    
+  
+    #Scikit Learn Model 
+    if len(cat_features) != 0:
+        X_sk = data_frame[con_features + cat_features]
+        y_sk = data_frame[response]
+        preprocessor = make_column_transformer((OneHotEncoder(handle_unknown="ignore"), cat_features))
+        pipeline = make_pipeline(preprocessor, PoissonRegressor(alpha=alpha, fit_intercept=True,))
+        sk_model = pipeline.fit(X_sk,y_sk)
+    else:
+        X_sk = data_frame[con_features]
+        y_sk = data_frame[response]
+        pipeline = make_pipeline(PoissonRegressor(alpha=0, fit_intercept=True, max_iter=100))
+        sk_model = pipeline.fit(X_sk,y_sk)    
+    #Aditive inferential model 
+    if model == "additive":
+        cat_features =["C(" + i + ")" for i in cat_features]
+        con_list = "".join([f"{i}" if i is con_features[0] else f" + {i}"for i in con_features])
+        cat_list = "".join([f"{i}" if i is cat_features[0] else f" + {i}"for i in cat_features])
+        if len(cat_list) > 0:
+            formula =  f"{response} ~ {con_list} + {cat_list}"
+        else:
+            formula =  f"{response} ~ {con_list}"
+        glm_count= smf.glm(formula=formula, data=data_frame, family=sm.families.Poisson()).fit()
+        print(glm_count.summary())   
+    else :
+        cat_features =["C(" + i + ")" for i in cat_features]
+        con_list = "".join([f"{i}" if i is con_features[0] else f" + {i}"for i in con_features])
+        cat_list = "".join([f"{i}" if i is cat_features[0] else f" + {i}"for i in cat_features])
+        interact_list = "".join([f"{i} * {j}" if j is cat_features[0] and i is con_features[0] 
+                         else f" + {i} * {j}" for i in con_features for j in cat_features])
+        equal = set()
+        cont_interaction = ""
+        for i in con_features[0:]:
```

Comment:
A three-level nested for-loop is a bit eye-catching. Is there any way to simplify this?